### PR TITLE
clippy: -D unnecessary_cast

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This pull request updates the tock source so that we pass the `unnecessary_cast` lint.

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: arch/cortex-m/src/mpu.rs [hunk 1]
```diff
@@ -472,7 +472,7 @@ impl<const NUM_REGIONS: usize, const MIN_REGION_SIZE: usize> mpu::MPU
                 let tz = start.trailing_zeros();
                 if tz < 32 {
                     // Find the largest power of two that divides `start`
-                    (1 as usize) << tz
+                    1_usize << tz
```
Comment (ppannuto) [inline arch/cortex-m/src/mpu.rs:475]: I acknowledge that this is presumably the more idiomatic Rust, and fine, whatever, we can write this way, but wow this is much less clear to my eyes.

File: boards/qemu_rv32_virt/src/io.rs [hunk 2]
```diff
@@ -8,7 +8,6 @@ use core::str;
 
 use kernel::debug;
 use kernel::debug::IoWrite;
-use rv32i;
```
Comment (ppannuto) [inline boards/qemu_rv32_virt/src/io.rs:11]: Weird this was caught in this particular lint, but w/e I guess

File: capsules/extra/src/touch.rs [hunk 3]
```diff
@@ -177,7 +177,7 @@ impl<'a> Touch<'a> {
                 }
                 ScreenRotation::Rotated270 => {
                     mem::swap(&mut width, &mut height);
-                    (width as u16 - touch_event.y as u16, touch_event.x)
+                    (width as u16 - touch_event.y, touch_event.x)
```
Comment (ppannuto) [inline capsules/extra/src/touch.rs:180]: This one is a little weirder as I think `touch_event.y` is still being coerced to a `u16` but _only_ becuase the `width` in the LHS has been coerced. This feels fragile. Maybe better to split into separate lines / temporary variables?

**Review by ppannuto — CHANGES_REQUESTED** [review]
I swear once upon a compiler a lot of these were required.

I kinda liked the intentionality of type conversion Rust asked for once upon a time, but the ergonomics of implicit conversion of unambiguous casts is admittedly nice.

File: capsules/extra/src/touch.rs (hunk 3 above)
Comment (lschuermann) [reply capsules/extra/src/touch.rs:180]: Seems that `touch_event.y` is actually a `u16`: https://github.com/tock/tock/blob/0a2b74eba355dba42907fb6f5d693166ad4539be/kernel/src/hil/touch.rs#L35

AFAIK, Rust (still) does not coerce types.

**Comment by lschuermann** [comment]
> I kinda liked the intentionality of type conversion Rust asked for once upon a time, but the ergonomics of implicit conversion of unambiguous casts is admittedly nice.

I could be wrong, I don't think that's what's happening here. Rust only implicitly assumes an integer time for literals which aren't otherwise "driven to a type" (e.g. through the `1_usize`). I believe many of these cases are just ones where we've changed the actual type underneath, such that our casts have literally become unnecessary.

**Review by lschuermann — APPROVED** [review]

File: boards/qemu_rv32_virt/src/io.rs (hunk 2 above)
Comment (bradjc) [reply boards/qemu_rv32_virt/src/io.rs:11]: Yeah it wasn't, clippy fix sometimes deletes a bunch of imports from some board's io.rs file. When I fixed this one I only put back what was necessary.

I don't really understand why io.rs and rust behave so strangely.

**Review by hudson-ayers — APPROVED** [review]

File: arch/cortex-m/src/mpu.rs (hunk 1 above)
Comment (brghena) [reply arch/cortex-m/src/mpu.rs:475]: I think it's the underscore that makes this seem weird to me. Would it be more clear without the underscore @ppannuto?

```rust
    1usize << tz
```

versus

```rust
    1_usize << tz
```

Comment (brghena) [reply arch/cortex-m/src/mpu.rs:475]: But we should really be consistent. So if `0x8000000_u32` and `0b01_u32` seem like the clear way to write them (and I think the underscore _is_ better there), then `1_usize` is for the best.

**Review by brghena — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]

**Pull request merged**